### PR TITLE
✨ feat(rust/sorting) add insertion, merge, quick sort code 

### PR DESCRIPTION
**Pull request by xBLACKICEx**
If this PR is related to coding or code translation, please read the [contribution guideline](https://github.com/krahets/hello-algo/issues/15#issue-1464069591), fill out the checklist, and paste the console outputs to the PR.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).
image

### CODE REVIEW

**Comment by krahets** [comment]
To be reviewed by @sjinzh

File: codes/rust/chapter_sorting/insertion_sort.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/*
+ * File: insertion_sort.rs
+ * Created Time: 2023-02-13
+ * Author: xBLACKICEx (xBLACKICEx@outlook.com)
+ */
+include!("../include/include.rs");
```
Comment (sjinzh) [inline codes/rust/chapter_sorting/insertion_sort.rs:6]: >```
>/*
> * File: insertion_sort.rs
> * Created Time: 2023-02-13
> * Author: xBLACKICEx (xBLACKICEx@outlook.com)
> */
>
>include!("../include/include.rs");
>```

File: codes/rust/chapter_sorting/merge_sort.rs [hunk 2]
```diff
@@ -0,0 +1,56 @@
+/**
+ * File: merge_sort.swift
```
Comment (sjinzh) [inline codes/rust/chapter_sorting/merge_sort.rs:2]: >```
> * File: merge_sort.rs
>```

File: codes/rust/chapter_sorting/merge_sort.rs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        }
+        // 否则，若“左右子数组都未全部合并完”且“左子数组元素 > 右子数组元素”，则选取右子数组元素，并且 j++
+        else {
+            nums[k] = tmp[r_corrent];
+            r_corrent += 1;
+        }
+    }
+}
+
+fn merge_sort(left: usize, right: usize, nums: &mut [i32]) {
```
Comment (sjinzh) [inline codes/rust/chapter_sorting/merge_sort.rs:39]: >```
>/* 归并排序 */
>fn merge_sort(left: usize, right: usize, nums: &mut [i32]) {
>```

File: codes/rust/chapter_sorting/quick_sort.rs [hunk 4]
```diff
@@ -0,0 +1,146 @@
+/**
+ * File: quick_sort.cpp
```
Comment (sjinzh) [inline codes/rust/chapter_sorting/quick_sort.rs:2]: >```
> * File: quick_sort.rs
>```

File: codes/rust/chapter_sorting/quick_sort.rs [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+ */
+
+// 快速排序
+struct QuickSort;
+// 快速排序（中位基准数优化）
+struct QuickSortMedian;
+// 快速排序（尾递归优化）
+struct QuickSortTailCall;
+
+/* 快速排序 */
```
Comment (sjinzh) [inline codes/rust/chapter_sorting/quick_sort.rs:14]: /* 快速排序类 */

File: codes/rust/chapter_sorting/quick_sort.rs [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+        // 哨兵划分
+        let pivot = Self::partition(nums, left as usize, right as usize) as i32;
+        // 递归左子数组、右子数组
+        Self::quick_sort(left, pivot - 1, nums);
+        Self::quick_sort(pivot + 1, right, nums);
+    }
+}
+
+/* 快速排序（中位基准数优化） */
+impl QuickSortMedian {
```
Comment (sjinzh) [inline codes/rust/chapter_sorting/quick_sort.rs:47]: /* 快速排序类（中位基准数优化） */

File: codes/rust/chapter_sorting/quick_sort.rs [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+        }
+        // 哨兵划分
+        let pivot = Self::partition(nums, left as usize, right as usize) as i32;
+        // 递归左子数组、右子数组
+        Self::quick_sort(left, pivot - 1, nums);
+        Self::quick_sort(pivot + 1, right, nums);
+    }
+}
+
+/* 快速排序（尾递归优化） */
```
Comment (sjinzh) [inline codes/rust/chapter_sorting/quick_sort.rs:94]: /* 快速排序类（尾递归优化） */

**Review by sjinzh — CHANGES_REQUESTED** [review]

File: codes/rust/chapter_sorting/quick_sort.rs (hunk 5 above)
Comment (xBLACKICEx) [reply codes/rust/chapter_sorting/quick_sort.rs:14]: rust 里没有类所以不写“类”，参考go语言

**Review by sjinzh — APPROVED** [review]
@krahets  It's OK.

**Review by krahets — APPROVED** [review]

**Pull request merged**